src/main/java/hudson/remoting/Launcher.java

```diff
@@ -465,6 +473,12 @@ private SSLSocketFactory getSSLSocketFactory()
      * Parses the connection arguments from JNLP file given in the URL.
      */
     public List<String> parseJnlpArguments() throws ParserConfigurationException, SAXException, IOException, InterruptedException {
+        if (secretFile != null) {
+            if (slaveJnlpCredentials != null || secret != null) {
+                throw new IOException("-jnlpCredentials, -secret, and -secretFile are mutually exclusive");
```

Comment:
also needs to update line 485:

    throw new IOException("-jnlpCredentials and -secret are mutually exclusive");

Perhaps determine separately how many of these are set and fail there rather that repeat checks?

Reply:
Yeah, I thought about that, but I figured it was easier to add and about as easy to read by just adding the one check. If secretFile is null then the message in 485 is still valid.

Reply:
Well, it should mention `secretFile` though.